internal/lint/check_file_options_unset.go

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package lint
+
+import (
+	"text/scanner"
+
+	"github.com/emicklei/proto"
+	"github.com/uber/prototool/internal/text"
+)
+
+var fileOptionsUnsetJavaMultipleFilesChecker = NewAddChecker(
+	"FILE_OPTIONS_UNSET_JAVA_MULTIPLE_FILES",
+	`Verifies that the file option "java_multiple_files" is unset.`,
+	newCheckFileOptionsUnset("java_multiple_files"),
+)
+
+var fileOptionsUnsetJavaOuterClassnameChecker = NewAddChecker(
+	"FILE_OPTIONS_UNSET_JAVA_OUTER_CLASSNAME",
+	`Verifies that the file option "java_outer_classname" is unset.`,
+	newCheckFileOptionsUnset("java_outer_classname"),
+)
+
+func newCheckFileOptionsUnset(fileOption string) func(func(*text.Failure), string, []*proto.Proto) error {
+	return func(add func(*text.Failure), dirPath string, descriptors []*proto.Proto) error {
+		return runVisitor(&fileOptionsUnsetVisitor{
+			baseAddVisitor: newBaseAddVisitor(add),
+			fileOption:     fileOption,
+		}, descriptors)
+	}
+}
+
+type fileOptionsUnsetVisitor struct {
+	baseAddVisitor
+
+	fileOption string
+
+	seen     bool
+	position scanner.Position
+}
+
+func (v *fileOptionsUnsetVisitor) OnStart(descriptor *proto.Proto) error {
+	v.seen = false
+	v.position = scanner.Position{}
```

Comment:
I don't know if these are actually necessary. `false` and `scanner.Position` should both be the default values for `*fileOptionsUnsetVisitor`. I think we can omit the `OnStart` method entirely (unless it's required to implement some interface. In that case, it could be:

```
func (v *fileOptionsUnsetVisitor) OnStart(descriptor *proto.Proto) error { return nil }
```

Reply:
Right now we re-use these linters, which admittedly isn’t a great pattern, so they need to be reset each time.

Reply:
Ok, good to know 👍

Reply:
This is unfortunate that linters are reused, but I can understand that it may not make much sense to rebuild them for every file -- @peter-edge you didn't by chance do any benchmarking on this did you :D

Reply:
No, but the time this all takes is very trivial, like on the order of milliseconds